pravega/flink-connectors
# [Issue-445] Drop travis build and use github actions for CI pipeline

Signed-off-by: zhongle.wang <zhongle_wang@dell.com>

**Change log description**

* Delete .travis.yml as we will no longer use Travis CI.
* Set up GitHub Actions workflow as follows:
    * Build and run tests on every push and pull requests.
    * Publish the snapshot **ONLY** on the push event of the master branch or the release branch (starts with `r0.x` or `r1.x`).
* Bonus features:
    * Gradle packages are cached, *but for for now we see no speed up ):*.
    * Workflow could be manually triggered via the Github Actions page, as long as the workflow file is in the master or default branch.

**Purpose of the change**

Fixes #445 

**What the code does**

When there is a push or a pull request, trigger the ci/cd to build and run tests.
The process will be:
* Clone the repository.
* Setup the JDK.
* Get the cache, or download the new packages if there is a update in `gradle.properties`.
* Grant execute permission for gradlew.
* Build via Gradle
* Build via Gradle with Scala 2.11
* Publish the snapshot if it is a push on the master or the release branch (starts with r0.x or r1.x).
* Publish the snapshot with Scala 2.11 if it is a push on the master or the release branch (starts with r0.x or r1.x).

**How to verify it**

Please verify that this is functionally equivalent to the previous CI pipeline:

* It compiles the code.
* It executes all unit tests and all integration tests.
* It uploads code coverage correctly.
* For master and release branches, it publishes to JFrog artifactory **ONLY** on push event.
* Triggers are correct and make sense.
